process/profile.go

```diff
@@ -82,10 +82,19 @@ func (p *Process) GetProfile(ctx context.Context) (changed bool, err error) {
 		}
 	}
 
-	// Get the (linked) local profile.
-	localProfile, err := profile.GetProfile(profile.SourceLocal, profileID, p.Path, false)
-	if err != nil {
-		return false, err
+	var localProfile *profile.Profile
+	if p.EnvironmentProfileID != "" && profileID == "" {
+		// Get the user specified profile.
+		localProfile, err = profile.GetProfile(profile.SourceEnvironmentDefined, p.EnvironmentProfileID, p.EnvironmentProfileID, false) // not setting process path since the profile an be used by multiple apps
```

Comment:
Do use `EnvironmentProfileID` as the `linkedPath` - it's not a path. ;)
It is not required anyway.